scripts/west_commands/runners/jlink.py

```diff
@@ -167,16 +174,20 @@ def do_run(self, command, **kwargs):
             client_cmd = (self.gdb_cmd +
                           self.tui_arg +
                           [self.elf_name] +
-                          ['-ex', 'target remote :{}'.format(self.gdb_port)])
+                          ['-ex', 'target remote {}:{}'.format(self.gdb_host, self.gdb_port)])
             if command == 'debug':
                 client_cmd += ['-ex', 'monitor halt',
                                '-ex', 'monitor reset',
                                '-ex', 'load']
                 if self.reset_after_load:
                     client_cmd += ['-ex', 'monitor reset']
-
-            self.print_gdbserver_message()
-            self.run_server_and_client(server_cmd, client_cmd)
+            if not self.gdb_host:
```

Comment:
This appears to be mixing up the west `debug` versus `debugserver` commands.

I guess that the presence of the `--gdb-host` command line option should mean that we are running `attach` or `debug` only, but the code isn't checking for this. Am I missing something?

Reply:
It is probably mixing both commands, but I didn't know how to do this in a clean way. 

But if the `--gdb-host` isn't `localhost` we can't run the server command anyway. Maybe a new command `debugclient` would be more appropriate?

Reply:
If I understand your purpose correctly, I think that checking the command is either `debug` or `attach` if `--gdb-host` is given and giving a fatal error message if it is not will suffice, right?

Reply:
Looking at the handling again, the cases for `flash` and `debugserver` are already checked, so we can only have `debug` or `attach` left.
So maybe it makes more sense to check if no `--gdb-host` is passed when running `debugserver`? Otherwise I don't see the added value.

Reply:
Sure, that sounds good.